src/libraries/System.Private.CoreLib/src/System/IO/StreamReader.cs

```diff
@@ -289,23 +289,47 @@ private void CloseStreamFromDispose(bool disposing)
                 base.DisposeAsync() :
                 DisposeAsyncCore();
 
-        private ValueTask DisposeAsyncCore()
+        private async ValueTask DisposeAsyncCore()
         {
-            // Same logic as in Dispose().
+            // Same logic as in StreamWriter.DisposeAsync().
             Debug.Assert(GetType() == typeof(StreamReader));
             try
             {
                 if (!_disposed)
                 {
+                    ThrowIfDisposed();
```

Comment:
Why if `_disposed` is checked right above?

Reply:
This is just mimicking the Writer Logic